program/tests/processor.rs

```diff
@@ -27,87 +27,114 @@ use {
             set_authority, sync_native, thaw_account, transfer, transfer_checked,
             ui_amount_to_amount, AuthorityType, MAX_SIGNERS,
         },
-        processor::Processor,
         state::{Account, AccountState, Mint, Multisig},
     },
-    std::sync::{Arc, RwLock},
+    std::{
+        collections::HashMap,
+        sync::{Arc, RwLock},
+    },
 };
 
 lazy_static::lazy_static! {
     static ref EXPECTED_DATA: Arc<RwLock<Vec<u8>>> = Arc::new(RwLock::new(Vec::new()));
 }
 
-fn set_expected_data(expected_data: Vec<u8>) {
-    *EXPECTED_DATA.write().unwrap() = expected_data;
-}
-
-struct SyscallStubs {}
-impl solana_sdk::program_stubs::SyscallStubs for SyscallStubs {
-    fn sol_log(&self, _message: &str) {}
-
-    fn sol_invoke_signed(
-        &self,
-        _instruction: &Instruction,
-        _account_infos: &[AccountInfo],
-        _signers_seeds: &[&[&[u8]]],
-    ) -> ProgramResult {
-        Err(ProgramError::Custom(42)) // Not supported
-    }
-
-    fn sol_get_clock_sysvar(&self, _var_addr: *mut u8) -> u64 {
-        program_error::UNSUPPORTED_SYSVAR
-    }
+fn do_process_instruction(
+    instruction: Instruction,
+    mut accounts: Vec<&mut SolanaAccount>,
+) -> ProgramResult {
+    let mut instruction_accounts = Vec::new();
 
-    fn sol_get_epoch_schedule_sysvar(&self, _var_addr: *mut u8) -> u64 {
-        program_error::UNSUPPORTED_SYSVAR
-    }
+    // Prepare accounts for mollusk.
+    instruction
+        .accounts
+        .iter()
+        .zip(&accounts)
+        .map(|(account_meta, account)| {
+            (
+                account_meta.pubkey,
+                AccountSharedData::from((*account).clone()),
+            )
+        })
+        .for_each(|(pubkey, account)| {
+            instruction_accounts.push((pubkey, account));
+        });
```

Comment:
nit: it's more idiomatic to do a `collect` rather than create the vec and push to it with a `for_each`

Reply:
💯